app/scripts/loadout/dimLoadout.directive.js

```diff
@@ -56,19 +56,13 @@
     function Link(scope) {
       var vm = scope.vm;
 
-      vm.classTypeValues = [{
-        label: 'Any',
-        value: -1
-      }, {
-        label: 'Warlock',
-        value: 0
-      }, {
-        label: 'Titan',
-        value: 1
-      }, {
-        label: 'Hunter',
-        value: 2
-      }];
+      vm.classTypeValues = [{ label: $translate.instant('Loadouts.Any'), value: -1 }];
+
+      scope.$on('dim-stores-updated', function(evt, data) {
+        _.each(_.reject(data.stores, function(s) { return s.isVault; }), function(store) {
+          vm.classTypeValues.push({ label: store.className, value: store.classType });
```

Comment:
This seems like it'll keep getting longer and longer as `dim-stores-updated` keeps getting fired. Probably better to build a new list and assign it each time.